tests/integration/database-management.test.js

```diff
@@ -0,0 +1,87 @@
+const {createKeys} = require('../../src/clientManager');
+const {localSwarmHooks} = require('../shared/hooks');
+const {generateString} = require('../../src/utils');
+
+
+describe('database management', function () {
+
+    localSwarmHooks({createDB: false});
+
+    // ensure DB sizes > 4GiB does not cause issues for JS client
+    const testParams = {
+        databaseSize: 4294967296,
```

Comment:
Note we need to test >4294967296, not ==, so a larger number should be used.

Upon further review I'm not sure why I picked 4GiB. 8192TiB is the more interesting threshold (2^53, also known as `Number.MAX_SAFE_INTEGER`). This is mostly interesting because, for example, `Number.MAX_SAFE_INTEGER + 1 === Number.MAX_SAFE_INTEGER + 2` does actually evaluate to `true`.

However, this is still ensuring that the whole system supports reasonably sized DBs (i.e. >2^32 bytes) and thus I'm happy with the test continuing to exist.